models.py

```diff
@@ -100,17 +101,29 @@ def annotations_by_books_and_prints(self, group_amount=50):
 
         return (books,prints)
 
+    def format_trp_id(self, trp_id):
+        return '%04d' % int(trp_id)
 
     def _get_trp_id(self):
-        last_bio = Biography.objects.order_by('-trp_id')[0]
+        ordered_bios = Biography.objects.order_by('-trp_id')
+        index = 0
+        last_bio = ordered_bios[index]
+        while last_bio.trp_id != self.format_trp_id(last_bio.trp_id):
+            last_bio = ordered_bios[index+1]
         last_trp_id = int(last_bio.trp_id)
         new_trp_id = last_trp_id + 1
         return '%04d' % new_trp_id
 
     def save(self, *args, **kwargs):
         if not self.trp_id:
             self.trp_id = self._get_trp_id()
-        super(Biography, self).save(*args, **kwargs)
+        else:
+            self.trp_id = self.format_trp_id(self.trp_id)        
+        self.roles = self.roles.replace(",",";"); 
+        try:
+            super(Biography, self).save(*args, **kwargs)
+        except IntegrityError as e:
+            print e.message
```

Comment:
Yea we should probably do something to indicate that the edit didn't save. 

This only occurs in the edge case when the user enters a value for the trp_id that is not in use before formatting, for example '88', so django doesn't flag in the interface that the value isn't unique. However, once formatted to '0088', the id is no longer unique so when the id gets sent to the db an integrity error is thrown.  

Would it make sense, in this case, to catch the error and then try to re-save the entry with an auto-generated incremental value for the trp_id, as is done with new entries? I don't quite understand why users are allowed to enter their own values for the trp_id to begin with - that is what started the conversation with Elli yesterday about whether the trp_id was necessary at all, (which it sounds like it is due to its current use in annotations).

What are you thoughts on best practice here?